packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -236,42 +238,15 @@ export class QuoteRequestor {
         if (!_opts.txOrigin) {
             _opts.txOrigin = constants.NULL_ADDRESS;
         }
-        const rawQuotes = await this._getQuotesAsync<V4RFQIndicativeQuote>(
+        return this._fetchAndValidateIndicativeQuotes(
             makerToken,
             takerToken,
             assetFillAmount,
             marketOperation,
             comparisonPrice,
             _opts,
-            'indicative',
+            this._rfqtAssetOfferings,
         );
-
-        // validate
```

Comment:
good to separate out this logic, consider a `validateQuote` function (non-async)

Reply:
This is a bit trickier than I originally thought. It's straightforward to create two functions: `validateFirmQuote` and a `validateIndicativeQuote` - but gets a bit gnarly when trying to unify them into a single `validateQuote` fn. 

Thoughts on having two separate functions?

Reply:
Started down the unified path and got here: https://github.com/0xProject/protocol/commit/0255b4deb2cff4d608bab9df4e85666167da776d

But haven't found something that I'm happy with. Will look into this in a separate PR if I have time